code/modules/jobs/job_types/_job.dm

```diff
@@ -264,7 +267,7 @@
 	// being only limited by its spawn variance limit
 	if (proxy == src || ignore_self_limit || proxy.dynamic_spawn_group)
 		position_limit = INFINITY
-	return min(position_limit, max(spawn_group_minimum + proxy.dynamic_spawn_variance_limit + total_position_delta, 0))
+	return min(position_limit, max(ceil(spawn_group_total / spawn_group_sizes) + proxy.dynamic_spawn_variance_limit + total_position_delta, 0))
```

Comment:
divide by zero possibility?

Reply:
It could potentially happen if a new job was given a group with only a single person in it, but this is a compile time issue and it would be obvious as it would runtime as soon as you test it